File: csharp/account/event/AccountCreatedHandler.cs
namespace ex.account;

using scyna;

public class AccountCreatedHandler : Event.Handler<proto.AccountCreated>
{
    public override void Execute()
    {
        context.Info("Receive AccountCreated event");
        Console.WriteLine(data);
        try
        {
            context.ScheduleTask(Path.SEND_EMAIL_CHANNEL, DateTimeOffset.Now.ToUnixTimeSeconds(), 61,
                    new proto.SendEmailTask { Email = data.Email, Content = "Just say hello" }, 2);
        }
        catch (scyna.Error e) { Console.WriteLine(e); }
    }
}

File: go/account/event/account_created_handler.go
package event

import (
	"log"
	"time"

	proto "ex/account/proto"
	"ex/account/service"

	scyna "github.com/scyna/core"
)

func AccountCreatedHandler(ctx *scyna.Event, event *proto.AccountCreated) {
	ctx.Info("AccountCreated handler")
	log.Print(event)
	ctx.ScheduleTask(service.SEND_EMAIL_CHANNEL, time.Now(), 61, &proto.SendEmailTask{
		Email:   event.Email,
		Content: "Just say hello",
	}, 2)
}
